modules/imgproc/src/histogram.cpp

```diff
@@ -825,10 +825,14 @@ static bool ipp_calchist(const Mat &image, Mat &hist, int histSize, const float*
     if(accumulate)
         ihist.create(1, &histSize, CV_32S);
 
+    int _histSize = histSize;
+    if(histSize > ranges[0][1])
+        _histSize = (int)ranges[0][1];
```

Comment:
1. `histSize` and `ranges` are not dependent values:
    - histSize specifies number of bins (dst size)
    - 'ranges' specifies which src values put into bins.

2. This check ignores non-uniform case.

Reply:
I believe we can use this check to disable IPP:
```
    // IPP_DISABLE_HISTOGRAM - https://github.com/opencv/opencv/issues/11544
    if (uniform && (ranges[0][1] - ranges[0][0]) != histSize)
        return false;
```

Please add this and remove other changes from this file.

Reply:
I again confirmed all the logic of "calcHist" and understood what you are saying. 
I fix it as you say. It was a great learning experience for me. I am always grateful for your help.